Implemented `IntervalTree::find_mut`

veldsla opened the pull request:
I needed mutable access to the Interval's data in the tree:

`IntervalTree::find_mut` like `find` returns an `Iterator`, but uses `EntryMut` which allows mutable access to the data part of the node in the tree.

I expected a long fight with the borrow checker, but amazingly it didn't complain a lot.
Also fixed a clippy warning in `utils::Interval`

johanneskoester commented:
Thanks, nice work!

johanneskoester commented on src/data_structures/interval_tree.rs:
```diff
@@ -100,6 +100,69 @@ impl<'a, N: Ord + Clone + 'a, D: 'a> Iterator for IntervalTreeIterator<'a, N, D>
+    }
+
+    /// Get a reference to the interval for this entry
+    pub fn interval(&self) -> &'a Interval<N> {
```
On the road to 100% test coverage, could you additionally call this in your test case?

veldsla replied:
For some additional percent we could also refactor the tree invariant tests into `debug_assert!`'s? These should never be triggered if the tree is working correctly.

johanneskoester replied:
Yes, sounds good!

johanneskoester commented:
Do you want me to merge, or do you first want to alter the asserts?

veldsla commented:
I though the panics were in the tree logic, but they were actually in the tests. I converted them to regular asserts in fec524b.

Added two more tests. Let's see if we can get full coverage.

veldsla commented:
Picked the low hanging coverage fruit. If tests pass feel free to merge.
